### ISSUE TITLE
Interface improvement

### ISSUE THREAD

**Issue by spaar** [spaar/besiege-modloader#2]
The current interface is rather ugly, ideally a skin closer to the game should be provided.
Additionally filtering methods and a clear button for the console would be nice.
- [x] Looks, ideally a skin closer to the game
- [x] Filtering of console messages
- [x] Font
- [ ] Save window positions in Configuration

**Comment by ZiMMy** [comment]
Would a Consolas font be suitable for the console?

**Comment by spaar** [comment]
I think Consolas would work quite well. It's monospace and designed for code or code-like things.
Do you know anything about its availability? I'm pretty sure it's shipped with Windows, but is it also availble on Mac and Linux? Or does the game/Unity already contain it?

**Comment by ZiMMy** [comment]
It's pre-installed on Windows only. We can include it as a resource, though.

The game only contain Arial and GOST Common fonts, AFAIR. And I don't think there is a monospaced font available on all systems by default.

**Comment by spaar** [comment]
Can we include it though? Since it's by Microsoft, I'm not sure if its license would allow that. I realize it's unlikely to get any problems due to it but I'd still rather not violate any licenses with this project.

**Comment by ZiMMy** [comment]
Oh well, then we can use Inconsolata, for example, or any font from https://www.google.com/fonts
I like Droid Sans Mono and Cousine.

Is support of non-latin characters needed btw?

**Comment by spaar** [comment]
Inconsolata looks nice. Also, thanks for the link to Google Fonts, never knew that existed.

I don't think that's particularly important since Besiege doesn't support any language other than English itself at the moment.

**Comment by ZiMMy** [comment]
Do we need a feature to save window positions to the config?

**Comment by spaar** [comment]
I suppose that would be a nice-to-have. I wouldn't say it's too high on the priority list though, some other stuff would be first. I'll make a check-list in the top post and include it there.

**Comment by ZiMMy** [comment]
Actually, I think we need some sort of plan.
Let's add all critical and urgent things that must be done to the issues list and then you create a milestone (that includes them) for the next release. After implementing all of them and testing, release a new version.

Then begin implementing next new features.

**Comment by spaar** [comment]
Sounds good. I was actually thinking, what do you think about the next version being 0.2 instead of 0.1.4? It's starting to become a pretty big update and also introduces API changes for mod developers (while staying backwards-compatible for now).

**Comment by ZiMMy** [comment]
I'm for 0.2

**Comment by spaar** [comment]
Okay, that's at least 2/3, I'll call it that.

**Comment by spaar** [comment]
Do you think I should add this to the milestone? I think we can postpone pretty much all of this to a future version, the next update is already getting big and this isn't really critical.

**Comment by ZiMMy** [comment]
My list:
- https://github.com/spaar/besiege-modloader/issues/20 - critical
- https://github.com/spaar/besiege-modloader/issues/2 - only KeySettings window (make it more tidy and compact, I can do it), others later
- clear button for the console

Everything else isn't urgent and works fine enough.

**Comment by spaar** [comment]
Sounds good. So... should I add the issue to the milestone or not?

**Comment by ZiMMy** [comment]
This one? Probably not. We're only gonna solve a small part of it.

**Comment by L3tum** [comment]
I was currently coding on the console...should I make a separate issue for that?
Also, I'd like to chat on something else than GitHub, because the different chats are hard to follow and my E-Mail gets spammed^^

**Comment by L3tum** [comment]
Oh, and also, if I have time, I'll make a pr for the changes to the keys...I think. I have to look what you've done so far.

**Comment by ZiMMy** [comment]
Let's create a skype group?

**Comment by spaar** [comment]
Meh, does that mean I need to install Skype? ^^ But sure, sounds good.

**Comment by L3tum** [comment]
Yeah, skype group sounds good...at least better than GitHub...
Also, I'd like to make a rule that there are only pr in the morning and in the afternoon, and one per day there you upload everything... it's kinda hard because I updated my project, then next commit, updated it, next commit...^^

**Comment by spaar** [comment]
I don't like that idea at all. There should be one pull request per change or one per related group of changes, not one per day. With git it, new commits appearing while you work shouldn't be a huge problem. Unless the commit significantly altered the code you were working on, it should be easy to integrate. And if it did change the code you were working on significantly, moving it to another time wouldn't really change that either. You don't necessarily immediately need to merge/rebase after each commit, just before pushing/creating a PR.

**Comment by L3tum** [comment]
Yeah, but it's hard to stay up to date if you're not online for a day or so. 
Like..for example, I was working on the config, and then commit, and I have to check if I have already done this or if I need to merge it.

**Comment by spaar** [comment]
I can see how it becomes problematic if two people are working on the same feature at the same time. Still, I think a time limit is not at all a good way to deal with this.

Also, just wanted to say: I decided to add both of you as collaborators again. Nevertheless, please create a pull request for any significant changes so that at least one other person can look over the code. It's alright to just commit&push small changes though.

**Comment by L3tum** [comment]
Ok, thanks :) 
Yeah, it was the first idea that came up in my mind... It's hard to make something like this if you're not a company or something... 
Maybe no time limit, but to try to make most of the changes together? Like not 10 prs per hour, rather collect them together or something..

**Comment by L3tum** [comment]
@ZiMMy  maybe a Whatsapp group? Skype sucks my RAM out and together with Visual Studio my PC would get pretty slow(it's 4 years old^^)

**Comment by ZiMMy** [comment]
How about Raidcall or Teamspeak?

**Comment by spaar** [comment]
I would be fine with all options discussed so far, though I have never used Raidcall. But if it works, why not?
Another suggestion: Slack. I've never used that either but it seems to be quite popular with developers and apparently has nice GitHub integration. Might be worth a try.

**Comment by L3tum** [comment]
I don't know Raidcall^^ And Teamspeak could be a problem since we're all not native english speakers and probably hard to understand. 
Slack is as far as I can see very good. It seems that they also have an app, don't they?

**Comment by spaar** [comment]
Indeed, they have clients for iOS, Android, Windows and Mac.

**Comment by ZiMMy** [comment]
@L3tum We don't have to use voice communication, every client has a text chat.

**Comment by L3tum** [comment]
Yeah, but the TS client's text chat isn't very....good or much used^^

**Comment by spaar** [comment]
So, what's your favorite idea of these? They won't do much good until we actually do something ;)
My favorite would be Slack. As I said, I have no experience with it, but it looks pretty nice. Whatever one we try first, we can of course always switch again if we don't like it anyway.

**Comment by L3tum** [comment]
I would prefer Slack, too. I haven't got experience, too, but it looks pretty nice and capable. 
We could also write our own communication client ;)

**Comment by L3tum** [comment]
You have to provide a company name for slack.

**Comment by spaar** [comment]
Give me a few minutes, I'll set something up.

**Comment by spaar** [comment]
Alright, I set up a team. You'll need to send me your email address to which I should send an invitation. Don't really care how, you can post it here, PM in the forums or whatever.

**Comment by spaar** [comment]
Closing this, the interface is very much reworked at this point and looks great (IMO).
